src/nvim/event/libuv_process.c

```diff
@@ -51,12 +51,19 @@ int libuv_process_spawn(LibuvProcess *uvproc)
 
   if (!proc->in.closed) {
     uvproc->uvstdio[0].flags = UV_CREATE_PIPE | UV_READABLE_PIPE;
+#ifdef WIN32
+    uvproc->uvstdio[0].flags |= UV_OVERLAPPED_PIPE;
+#endif
     uvproc->uvstdio[0].data.stream = STRUCT_CAST(uv_stream_t,
                                                  &proc->in.uv.pipe);
   }
 
   if (!proc->out.closed) {
     uvproc->uvstdio[1].flags = UV_CREATE_PIPE | UV_WRITABLE_PIPE;
+#ifdef WIN32
+    // pipe must be readable for IOCP to work.
+    uvproc->uvstdio[1].flags |= UV_READABLE_PIPE | UV_OVERLAPPED_PIPE;
+#endif
```

Comment:
We will need to somehow feature-detect the presence of `UV_OVERLAPPED_PIPE` (or check libuv version-string, if they `#define` those). 

Example: https://github.com/justinmk/neovim/blob/0f9f46056f1f2937fe9d0dd2a13996181e360417/src/nvim/tui/input.c#L37-L44

Reply:
Ideally I would like Nvim 0.2.3+ builds on windows to always support overlapped mode. it will make pynvim significantly simpler, we don't need to manually check handle state and implement our own reader thread (ideally asyncio should do all this but it is what it is), we can just use the proactor loop as-is.